spec/amp-html-layout.md

```diff
@@ -0,0 +1,159 @@
+<!---
+Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS-IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# AMP HTML ⚡ Layout System
+
+## Overview
+
+The main goal of the layout system is to ensure that AMP elements can express their layout
+so that the runtime is able to infer sizing of elements before any remote reosurces, such as
+JavaScript and data calls, have been completed. This is important since this along
+significantly reduces rendering and scrolling jank.
+
+With this in mind the AMP Layout System is designed to support few but flexible layout
+that provide good performance guarantees. This system relies on a set of attributes such
+as `layout`, `width` and `height` to express the element's layout and sizing needs.
+
+## Layout Attributes
+
+### `width` and `height`
+
+Depending on the value of the `layout` attribute AMP component elements must have a `width` and
+`height` attribute that contains an integer pixel value. Actual layout behavior is determined by the
+`layout` attribute as described below.
+
+In a few cases if `width` or `height` are not specified the AMP runtime can default these values
+as following:
+- `amp-pixel`: Both `width` and `height` are defaulted to 0.
+- `amp-audio`: The default `width` and `height` are inferred from browser.
+
+### `layout`
+
+The optional layout attribute allows specifying how the component behaves in the document layout.
+Valid values for the layout attribute are:
+
+- Not present: The `layout` will be inferred as following:
+  - if `width` equals to `auto` `fixed-height` layout is assumed;
+  - if `width` or `height` attributes are present `fixed` layout is assumed;
+  - if `width` and `height` are not present `container` layout is assumed
+- `fixed`: The `width` and `height` attributes must be present. The only exceptions are `amp-pixel`
+and `amp-audio` elements.
+- `fixed-height`: The `height` attribute must be present. The `width` attribute must not be present
+or must be equal to `auto`.
+- `responsive`: The `width` and `height` attributes must be present and are used to determine the
+aspect ratio of the component and the component is sized to the width of its container element while
```

Comment:
'ratio of the component and the component', consider some form of separator here to make it a little more human readable.